packages/core/src/process/4-mutation-test-executor.ts

```diff
@@ -91,33 +91,76 @@ export class MutationTestExecutor {
   }
 
   private executeCheck(input$: Observable<MutantTestCoverage>) {
-    const checkTask$ = this.checkerPool
-      .schedule(input$, async (checker, mutant) => {
-        const checkResult = await checker.check(mutant);
-        return {
-          checkResult,
-          mutant,
-        };
+    if (!this.options.checkers.length) {
+      const checkResult$ = new Subject<MutantResult>();
+      checkResult$.complete();
+      return {
+        checkResult$: checkResult$.asObservable(),
+        passedMutant$: input$,
+      };
+    }
+
+    const checkResult$ = new Subject<MutantResult>();
+    let previousPassedMutants$ = input$;
+
+    for (const checkerType of this.options.checkers) {
+      const passedMutants$ = new Subject<MutantTestCoverage>();
+      this.executeChecker(checkerType, previousPassedMutants$, checkResult$, passedMutants$);
+      previousPassedMutants$ = passedMutants$;
+    }
+
+    previousPassedMutants$.subscribe({
+      complete: () => {
+        this.log.debug('Checker(s) finished.');
+
+        checkResult$.complete();
+
+        this.checkerPool.dispose();
+        this.concurrencyTokenProvider.freeCheckers();
+      },
+    });
+
+    return {
+      checkResult$: checkResult$.asObservable(),
+      passedMutant$: previousPassedMutants$,
+    };
+  }
+
+  private async executeChecker(
+    checkerType: string,
+    previousPassedMutants$: Observable<MutantTestCoverage>,
+    checkResult$: Subject<MutantResult>,
+    passedMutant$: Subject<MutantTestCoverage>
+  ) {
+    const mutants = await lastValueFrom(merge(previousPassedMutants$).pipe(toArray()));
```

Comment:
This is fine I think for now. We might need to group them using an operator like [`bufferTime`](https://rxjs.dev/api/operators/bufferTime) when we have support more checker implementations.